src/coreclr/vm/interpexec.cpp

```diff
@@ -2121,19 +2161,56 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                 case INTOP_CALL_HELPER_V_AGS:
                 {
                     InterpGenericLookup *pLookup = (InterpGenericLookup*)&pMethod->pDataItems[ip[5]];
-                    void* helperArg = DoGenericLookup(LOCAL_VAR(ip[2], void*), pLookup);
+                    void* helperArg1 = LOCAL_VAR_ADDR(ip[1], void*);
+                    void* helperArg2 = DoGenericLookup(LOCAL_VAR(ip[2], void*), pLookup);
+                    void* helperArg3 = LOCAL_VAR(ip[3], void*);
+
+                    MethodDesc *pILTargetMethod = NULL;
+                    HELPER_FTN_V_PPP helperFtn = GetPossiblyIndirectHelper<HELPER_FTN_V_PPP>(pMethod, ip[4], &pILTargetMethod);
+                    if (pILTargetMethod != NULL)
+                    {
+                        returnOffset = -1;
```

Comment:
If I'm not mistaken, a lot of existing code just does `stack + returnOffset` without checking whether it's -1 first, so this is unsafe

Reply:
My intention here is to rather fail quickly when we try to read or write from/to return address for methods with void return value.

Another option is to set it to `ip[1]` as we do elsewhere, even when calling void methods.

Reply:
I think instead of failing quickly, on 64-bit targets it's possible it would just write to a random address in memory ~4GB away from the interpreter stack. I would prefer if we had a way to make sure the return address is NULL or a safe location to write to.

Reply:
And I just realized that depending on how the stack math is done, it would write a few bytes *before* the interpreter stack, which will almost certainly work and trash heap internal data structures.